File: testsuite/octave_modular/util/check_accuracy.m
function y = check_accuracy(accuracy, data)
	ending=sprintf('<--- accuracy: %e\n', accuracy);
	if strcmp(data{1}, 'kernel')==1 || strcmp(data{1}, 'distance')==1
		fprintf('train: %e, test: %e %s', data{2}, data{3}, ending);
	elseif strcmp(data{1}, 'classifier')==1
		fprintf('alphas: %e, bias: %e, sv: %e, classified: %e %s', ...
			data{2}, data{3}, data{4}, data{5}, ending);
	elseif strcmp(data{1}, 'kmeans')==1
		fprintf('centers: %e, radi: %e %s', data{2}, data{3}, ending);
	elseif strcmp(data{1}, 'hierarchical')==1
		fprintf('merge_distances: %e, pairs: %e %s', ...
			data{2}, data{3}, ending);
	elseif strcmp(data{1}, 'distribution')==1
		fprintf('likelihood: %e, derivatives: %e %s', ...
			data{2}, data{3}, ending);
	elseif strcmp(data{1}, 'custom')==1
		fprintf('triangletriangle: %e, fulltriangle: %e, fullfull: %e %s', ...
			data{2}, data{3}, data{4}, ending);
	end

	y=true;
	for i = 2:length(data)
		if data{i}>accuracy
			y=false;
			return;
		end
	end

File: testsuite/r_static/util/check_accuracy.R
check_accuracy <- function(accuracy, type, data) {
	if (regexpr(type, 'classifier')>0) {
		print(paste('alphas:', data[1], ', bias:', data[2], ', sv:', data[3], ', classified:', data[4], '<--- accuracy ', accuracy))
	} else if (regexpr(type, 'distance')>0) {
		print(paste('dtrain:', data[1], ', dtest:', data[2], '<--- accuracy ', accuracy))
	} else if (regexpr(type, 'distribution')>0) {
		print(paste('likelihood:', data[1], ', derivatives:', data[2], '<--- accuracy ', accuracy))
	} else if (regexpr(type, 'hierarchical')>0) {
		print(paste('merge_distances:', data[1], ', pairs:', data[2], '<--- accuracy', accuracy))
	} else if (regexpr(type, 'kernel')>0) {
		print(paste('ktrain:', data[1], ', ktest:', data[2], '<--- accuracy', accuracy))
	} else if (regexpr(type, 'kmeans')>0) {
		print(paste('centers:', data[1], ', radi:', data[2], '<--- accuracy', accuracy))
	} else {
		print(paste('Unknown accuracy type', type, '!'))
		return(FALSE)
	}

	for (i in 1:length(data)) {
		if (data[i]>accuracy) {
			return(FALSE)
		}
	}

	return(TRUE)
}
